ReloadExtension.js

```diff
@@ -0,0 +1,237 @@
+    }
+}(this, function(cometdModule) {
+    /**
+     * The reload extension allows a page to be loaded (or reloaded)
+     * without having to re-handshake in the new (or reloaded) page,
+     * therefore resuming the existing CometD connection.
+     *
+     * When the reload() method is called, the state of the CometD
+     * connection is stored in the window.sessionStorage object.
+     * The reload() method must therefore be called by page unload
+     * handlers, often provided by JavaScript toolkits.
+     *
+     * When the page is (re)loaded, this extension checks the
+     * window.sessionStorage and restores the CometD connection,
+     * maintaining the same CometD clientId.
+     */
+    return cometdModule.ReloadExtension = function(configuration) {
+        var _cometd;
+        var _debug;
+        var _state = {};
+        var _name = 'org.cometd.reload';
+        var _batch = false;
+        var _reloading = false;
+
+        function _reload(config) {
+            if (_state.handshakeResponse) {
+                _reloading = true;
+                var transport = _cometd.getTransport();
+                if (transport) {
+                    transport.abort();
+                }
+                _configure(config);
+                var state = JSON.stringify(_state);
+                _debug('Reload extension saving state', state);
+                window.sessionStorage.setItem(_name, state);
+            }
+        }
+
+        function _similarState(oldState) {
+            // We want to check here that the CometD object
+            // did not change much between reloads.
+            // We just check the URL for now, but in future
+            // further checks may involve the transport type
+            // and other configuration parameters.
+            return _state.url == oldState.url;
+        }
+
+        function _configure(config) {
+            if (config) {
+                if (typeof config.name === 'string') {
+                    _name = config.name;
+                }
+            }
+        }
+
+        function _receive(response) {
+            _cometd.receive(response);
+        }
+
+        this.configure = _configure;
+
+        this._receive = _receive;
+
+        this.registered = function(name, cometd) {
+            _cometd = cometd;
+            _cometd.reload = _reload;
+            _debug = _cometd._debug;
+        };
+
+        this.unregistered = function() {
+            delete _cometd.reload;
+            _cometd = null;
+        };
+
+        this.outgoing = function(message) {
+            switch (message.channel) {
+                case '/meta/handshake':
+                {
+                    _state = {};
+                    _state.url = _cometd.getURL();
+
+                    var state = window.sessionStorage.getItem(_name);
+                    _debug('Reload extension found state', state);
+                    // Is there a saved handshake response from a prior load ?
```

Comment:
OMG, there should be no space before a sentence and and ending punctuation mark. EVER!